backend-server/src/main/java/org/dice/ida/action/process/ActionExecutor.java

```diff
@@ -0,0 +1,43 @@
+package org.dice.ida.action.process;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.dice.ida.action.def.Action;
+import org.dice.ida.constant.IDAConst;
+import org.dice.ida.model.ChatMessageResponse;
+import org.dice.ida.model.Intent;
+
+import com.google.cloud.dialogflow.v2.QueryResult;
+public class ActionExecutor {
+
+	private Action action;
+	private Map<String, Object> paramMap;
+
+	public ActionExecutor(QueryResult queryResult) {
+		// Initiate the instance for the action
+		Intent intent = Intent.getForKey(queryResult.getIntent().getDisplayName());
+		this.paramMap = createParamMap(queryResult);
+		this.action = ActionMappingHelper.fetchActionInstance(intent.getKey());
+	}
+
+	private Map<String, Object> createParamMap(QueryResult queryResult){
+
+		Map<String, Object> paramMap = new HashMap<>();
+		String messageResponseText = queryResult.getFulfillmentText();
+
+		paramMap.put(IDAConst.PARAM_TEXT_MSG, messageResponseText);
+		paramMap.put(IDAConst.PARAM_ALL_REQUIRED_PARAMS_PRESENT, queryResult.getAllRequiredParamsPresent());
+
+		if (queryResult.getParameters().getFieldsCount() > 0) {
+			paramMap.put(IDAConst.PARAM_DATASET_NAME, queryResult.getParameters().getFieldsMap().get("datasetname").getStringValue());
```

Comment:
I think it'd be a good idea to extract parameters in the intent specific executor. For example, the dataset name might not be part of any other intent except for the load dataset.

Reply:
I suggest extracting all the parameters at a common place. Since we do not want to pass too much information to the specific `Action` instance. However, the way its done here is not right. Ideally one should write everything in `queryResult.getParameters()` into the `paramMap`.